<launch>
    <arg name="bag_file" default="" />
    <arg name="results_file" default="$(arg bag_file).reconstruction.hdf5" />
    <arg name="camera_model" default="double_sphere" />
    <arg name="camera_params" default="{fx: 295.936, fy: 295.936, cx: 512, cy: 512, chi: 0.3, alpha: 0.6666667}" />
    <arg name="rate" default="1" />
    <node pkg="omni_slam_eval" type="omni_slam_reconstruction_eval_node" name="omni_slam_reconstruction_eval_node" required="true" output="screen">
        <param name="bag_file" value="$(arg bag_file)" />
        <param name="results_file" value="$(arg results_file)" />
        <param name="image_topic" value="/unity_ros/Sphere/FisheyeCamera/image_raw" />
        <param name="depth_image_topic" value="/unity_ros/Sphere/FisheyeDepthCamera/image_raw" />
        <param name="pose_topic" value="/unity_ros/Sphere/TrueState/pose" />
        <param name="tracked_image_topic" value="/omni_slam_eval/tracked" />
        <param name="rate" value="$(arg rate)" />
        <rosparam subst_value="true">
            camera_model: '$(arg camera_model)'
            camera_parameters: $(arg camera_params)
            detector_type: 'GFTT'
            detector_parameters: {maxCorners: 100, qualityLevel: 0.01, minDistance: 5, blockSize: 5}
            tracker_window_size: 128
            tracker_num_scales: 4
            tracker_delta_pixel_error_threshold: 3.0
            tracker_error_threshold: 20.0
            min_features_per_region: 10
            max_reprojection_error: 0.0
            min_triangulation_angle: 5.0
        </rosparam>
    </node>
</launch>
